File: 8kyu/if-you-cant-sleep-just-count-sheep/if-you-cant-sleep-just-count-sheep.py
# def count_sheep(n):
#   res = ""
#   for num in range(1, n + 1):
#     if num == 0:
#       return res
#     else:
#       res += "{} sheep...".format(num)
#   return res

def count_sheep(n):
  return "".join(f"{num} sheep..." for num in range(1, n + 1))


print("start")
print(count_sheep(0))
print(count_sheep(3))

File: 8kyu/if-you-cant-sleep-just-count-sheep/if-you-cant-sleep-just-count-sheep.go
package main

import (
	"fmt"
)

// func countSheep(num int) string {
//   res := ""
// 	for i := 1; i <= num; i++ {
// 		if i == 0 {
// 			return res
// 		}
// 		res += strconv.Itoa(i) + " sheep..."
// 	}

// 	return res
// }

func countSheep(num int) string {
  if num > 0 {
		return countSheep(num - 1) + fmt.Sprintf("%d sheep...", num)
	}

	return ""
}

func main() {
	fmt.Println(countSheep(0))
	fmt.Println(countSheep(1))
	fmt.Println(countSheep(3))
}